src/main/java/com/example/document/service/DocumentService.java

```diff
@@ -36,19 +36,12 @@ public List<DocumentWithEmployeeDTO> uploadExcelDocument(MultipartFile file) thr
 
         List<DocumentEntry> docEntryList = new ArrayList<>();
 
-//        Employee validation
-        for (int i = 1; i <= sheet.getLastRowNum(); i++) {
-            Row row = sheet.getRow(i);
-            Long employeeId = (long) row.getCell(2).getNumericCellValue();
-            employeeClient.getEmployeeById(employeeId);
-        }
-
         for (int i = 1; i <= sheet.getLastRowNum(); i++) {
             Row row = sheet.getRow(i);
             LocalDate uploadDate = row.getCell(0).getLocalDateTimeCellValue().toLocalDate();
             LocalDate effectiveDate = row.getCell(1).getLocalDateTimeCellValue().toLocalDate();
             Long employeeId = (long) row.getCell(2).getNumericCellValue();
-
+            employeeClient.getEmployeeById(employeeId);
```

Comment:
We also have a loop in another method if I am not mistaken
We should merge them into one too

Reply:
yes, you are right, I used for loop in the insertDocumentEntries method, but there is only one loop to validate employees and then the provided list be saved.

So, in the second method,  there is no duplicated loop.